archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
-                                print("{} note deleted from object {}".format(
-                                    args.note_type, record.uri))
-                            if args.action == "modify":
-                                print("{} note was originally {} and was changed to {} in object {}".format(
-                                    args.note_type, content, args.replace_string, record.uri))
-                                subnote["content"] = args.replace_string
-                            log_to_spreadsheet(record)
-                            updated = True
-            if updated:
-                save_record(record.uri, aojson)
+                            handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated)
+                            
+def handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated):
+    if args.action == "delete":
+        del notes[idx]
+        print("{} note {} deleted from object {}".format(
+            args.note_type, args.search_string, record['uri']))
+    if args.action == "modify":
+        print("{} note was originally {} and was changed to {} in object {}".format(
+            args.note_type, content, args.replace_string, record['uri']))
+        subnote['content'] = args.replace_string
+    log_to_spreadsheet(record, writer, resource_id, client)
+    updated = True
+    if updated:
+        save_record(client, record['uri'], record)
 
 def contains_match(content, search_string):
     """Returns True if user-provided note input matches the corresponding note within a given ratio (CONFIDENCE_RATIO)."""
     ratio = fuzz.token_sort_ratio(content.lower(), search_string.lower())
     return True if ratio > CONFIDENCE_RATIO else False
 
-
-def save_record(uri, data):
+def save_record(client, uri, data):
     """Posts modifications/deletions to ArchivesSpace"""
-    updated = aspace.client.post(uri, json=data)
+    updated = client.post(uri, json=data)
     updated.raise_for_status()
 
-
-def log_to_spreadsheet(archival_object):
+def log_to_spreadsheet(record, writer, resource_id, client):
     """Logs top container identifier information of changed archival objects to spreadsheet"""
-    for instance in archival_object.instances:
-        top_container = instance.sub_container.top_container
-        container = "{} {}".format(
-            top_container.type.capitalize(), top_container.indicator)
-        writer.writerow([container, archival_object.uri])
+    if record['instances']:
+        for instance in record['instances']:
+            top_container = client.get(instance['sub_container']['top_container']['ref']).json()
+            container = "{} {}".format(
+            top_container['type'].capitalize(), top_container['indicator'])
+    else: 
+        container = "None"
+    writer.writerow([resource_id['title'], resource_id['id_0'], record['uri'], record['ref_id'], record['title'], container])
```

Comment:
```suggestion
    writer.writerow([resource_id['title'], resource_id['id_0'], record['uri'], record['ref_id'], record.get('title'), container])
```
I am pretty sure that it's possible for AS objects to not have a title if they have a date, so let's accommodate that edge case.